server/src/main/java/org/elasticsearch/search/fetch/PreloadedFieldLookupProvider.java

```diff
@@ -44,8 +54,23 @@ public void populateFieldLookup(FieldLookup fieldLookup, int doc) throws IOExcep
         backUpLoader.populateFieldLookup(fieldLookup, doc);
     }
 
+    void setPreloadedStoredFields(Set<String> preloadedStoredFields) {
+        this.preloadedStoredFields = preloadedStoredFields;
+    }
+
+    void setStoredFields(String id, Map<String, List<Object>> storedFields) {
```

Comment:
Maybe `setPreloadedStoredFieldValues` and the one above is `setPreloadedStoredFieldsNames`. That way it's clear these are mirror of eachother.